bundly/dash
# Server Re-write

- [x] Server setup
- [x] Schema added
- [x] Controllers added
- [x] GitHub OAuth2
- [x] Discord OAuth2
- <s>Google OAuth2</s> (pushed for further release, due to low priority)

## How to check
- Add the env variables (discord and github client keys)
- Go to `/server`
- Run `yarn install`
- `yarn start`
- For Github login
  - Open `http://localhost:5000/auth/github/login`
    - It should ask you your acc permissions
    - It returns success with access-token and your profile details
    - Store it with react in localStorage/Cookie idc
- For discord login
  - Open `http://localhost:5000/auth/discord/login`
  - Should throw error as github-access-token not passed (meaning user not logged in)
  - Try again passing with `http://localhost:5000/auth/discord/login?token=abcdefg`
  - Give access to discord
  - Should return an object with discord and github details combined

## Env variables sample

```
NODE_ENV=development

GITHUB_CLIENT_ID=abcd
GITHUB_CLIENT_SECRET=efgh

DISCORD_CLIENT_ID=hijk
DISCORD_CLIENT_SECRET=lmno
```
